x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1243,6 +1248,25 @@ private Expression resolveConvertFunction(AbstractConvertFunction convert, List<
             return convert;
         }
 
+        private static Expression resolveFoldablesConvertFunction(FoldablesConvertFunction fcf) {
+            if (fcf.field() instanceof FieldAttribute fa && fa.field() instanceof InvalidMappedField imf) {
```

Comment:
In the one place where this method is called, the field is always an `InvalidMappedField` so this check is redundant, isn't it?

Reply:
Yeah, this method is called in only one place, the checks here are used so that we don't have to do unnecessary casting in the error messages. I'll embed the error message in the caller in stead, so that we don't have to do unnecessary check or casting here.